EIPS/draft-factory-standard.md

```diff
@@ -0,0 +1,106 @@
+
+## Rationale
+- To save gas and storage space, the preamble should be as minimal as possible.
+
+- It is considered "bad" behavior to try to CALL a factory contract directly, therefore the preamble starts with `INVALID (0xfe)` to end execution with an exceptional halting condition (rather than a "gentler" opcode like `STOP (0x00)`).
+
+- To help distinguish a factory contract from other contracts that may start with `0xFE`, a "magic" byte is used. The value `0x71` was chosen by taking the last byte of the keccak256 hash of the bytestring "factory" (i.e.: `keccak256(b"factory")[-1]`).
+
+- An empty initcode is disallowed by the spec to prevent what might be a common mistake.
+
+- Users may want to include arbitrary data or code in their preamble. To allow indexers to ignore these bytes, a variable length encoding is proposed. To allow the length to be only zero or one bytes (in the presumably common case that `len(data bytes)` is smaller than 256), two bits of the third byte are reserved to specify how many bytes the encoded length takes.
+
+- In case we need an upgrade path, version bits are included. While we do not expect to exhaust the version bits, in case we do, a continuation sequence is reserved. Since only two bytes are required for `<length bytes>` (as [EIP-170](/EIPS/eip-170.md) restricts contract length to 24KB), a `<length encoding bits>` value of 3 would never be required to describe `<length bytes>`. For that reason, the special `<length encoding bits>` value of `0b11` is reserved as a continuation sequence marker.
+
+- The length of the initcode itself is not included by default in the preamble because it takes space, and it can be trivially determined using `EXTCODESIZE`.
+
+- The EOF ([EIP-3540](/EIPS/eip-3540.md)) could provide another way of specifying factory contracts, by adding another section kind (3 - initcode). However, it is not yet in the EVM, and we would like to be able to standardize factory contracts today, without relying on EVM changes. If, at some future point, section kind 3 becomes part of the EOF spec, and the EOF becomes part of the EVM, this ERC will be considered to be obsolesced since the EOF validation spec provides much stronger guarantees than this ERC.
+
+
+## Backwards Compatibility
+All EIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The EIP must explain how the author proposes to deal with these incompatibilities. EIP submissions without a sufficient backwards compatibility treatise may be rejected outright.
```

Comment:
```suggestion
TBD
```
Please don't leave placeholder text in.  Use `TBD` if you are planning on filling it in after getting the initial draft merged.  This makes it stand out more that it isn't done yet.